python/across_api/auth/api.py

```diff
@@ -56,17 +57,20 @@ async def __call__(self, request: Request) -> None:
                 resp = await client.get(jwks_uri)
             jwks_data = resp.json()
             header = jwt.get_unverified_header(credentials.credentials)
-            for signing_key in jwks_data["keys"]:
-                if signing_key["kid"] == header["kid"]:
-                    break
-            else:
+
+            signing_key = None
+            for key in jwks_data["keys"]:
+                if key["kid"] == header["kid"]:
+                    signing_key = key
+
+            if signing_key is None:
```

Comment:
Any reason to replace the `for/else` version of this?

Reply:
Honestly, I'd never used a `for/if-break/else` loop before, and I was experiencing a bug in testing. I changed it to this just in case.

For me it was more readable, and generally I find that instantiating a variable that is used later in the code as the for-loop key is hard to follow. 

I'll change it back to the for-else with this variation:
```python
signing_key = None
for key in jwks_data["keys"]:
    if key["kid"] == header["kid"]:
        signing_key=key
        break
else:
```

Reply:
I think this is fair enough, the for/else version was compact, but is more esoteric, and therefore less readable.